megamek/src/megamek/common/Entity.java

```diff
@@ -11887,9 +11887,7 @@ public void setGameOptions() {
         }
 
         for (Mounted mounted : getWeaponList()) {
-            if (mounted.getType() instanceof Weapon) {
-                ((Weapon) mounted.getType()).adaptToGameOptions(game.getOptions());
-            }
+            mounted.adaptToGameOptions(game.getOptions());
```

Comment:
I assume it's not possible for a mounted returned here to be anything other than a weapon?

Reply:
After the change it's calling adaptToGameOptions on mounted instead of mounted.getType(). Mounted now handles the weapon check, while the InfantryWeaponMounted subclass calls adaptToGameOptions on both weapons.